src/sentry/utils/dates.py

```diff
@@ -76,31 +76,27 @@ def parse_date(datestr: str, timestr: str) -> datetime | None:
             return None
 
 
-def parse_timestamp(value: Any) -> datetime | None:
+def parse_timestamp(value: datetime | int | float | str | bytes | None) -> datetime | None:
     # TODO(mitsuhiko): merge this code with coreapis date parser
-    if isinstance(value, datetime):
+    if not value:
+        return None
+    elif isinstance(value, datetime):
         return value
     elif isinstance(value, (int, float)):
         return datetime.fromtimestamp(value, UTC)
-    value = (value or "").rstrip("Z").encode("ascii", "replace").split(b".", 1)
-    if not value:
-        return None
+    elif isinstance(value, bytes):
+        value = value.decode()
+
     try:
         # Attempt to decode without timezone
-        rv = datetime.strptime(value[0].decode("ascii"), "%Y-%m-%dT%H:%M:%S")
-    except ValueError:
-        # Fallback to decoding with timezone as a ISO8610 string
+        rv = parse(value, ignoretz=True)
+    except (ParserError, ValueError):
+        # Fallback to decoding with timezone as a ISO-8601 string
         try:
-            rv = datetime.fromisoformat(value[0].decode("ascii"))
-        except Exception:
-            return None
-    except Exception:
-        return None
-    if len(value) == 2:
-        try:
-            rv = rv.replace(microsecond=int(value[1].ljust(6, b"0")[:6]))
-        except ValueError:
+            rv = isoparse(value)
+        except (ParserError, ValueError):
```

Comment:
I don't know if this is really safe or if we're relying on this in some way. How about we do something like
```
def parse_timestamp(value: datetime | int | float | str | bytes | None) -> datetime | None:
    ... # existing logic
    rv = _parse_timestamp_old(value)
    if options.get("some_flag"):
        new_value = parse(value, ignoretz=True)
        match = "equal"
        if new_value != rv:
            # Maybe log?
            match = "not_equal"
        metrics.incr("parse_date", tags={"result": match})
        rv = value
```

Reply:
Makes sense. I've updated the implementation. Thank you for the review Dan